tests/testthat/test-epichains.R

```diff
@@ -0,0 +1,206 @@
+set.seed(12)
+epichains_summary <- simulate_summary(
+  nchains = 10,
+  statistic = "size",
+  offspring_dist = "pois",
+  stat_max = 10,
+  lambda = 2
+)
+epichains_tree <- simulate_tree(
+  nchains = 10,
+  statistic = "size",
+  offspring_dist = "pois",
+  stat_max = 10,
+  lambda = 2
+)
+epichains_tree2 <- simulate_tree(
+  nchains = 10,
+  statistic = "size",
+  offspring_dist = "pois",
+  stat_max = 10,
+  serials_dist = function(x) 3,
+  lambda = 2
+)
+
+aggreg_by_gen <- aggregate(
+  epichains_tree,
+  grouping_var = "generation"
+)
+aggreg_by_time <- aggregate(
+  epichains_tree2,
+  grouping_var = "time"
+)
+
+aggreg_by_both <- aggregate(
+  epichains_tree2,
+  grouping_var = "both"
+)
+
+set.seed(11223)
+epichains_summary_all_infs <- simulate_summary(
+  nchains = 10,
+  statistic = "size",
+  offspring_dist = "pois",
+  stat_max = 10,
+  lambda = 3
+)
+
+test_that("print.epichains works for simulate_summary output", {
+  expect_snapshot(epichains_summary)
+})
+
+test_that("print.epichains works for simulate_tree output", {
+  expect_snapshot(epichains_tree)
+})
+
+test_that("print.epichains works for simulate_tree output", {
+  expect_snapshot(epichains_tree2)
+})
+
+test_that("summary.epichains works as expected", {
+  expect_named(
+    summary(epichains_summary),
+    c(
+      "chain_ran",
+      "max_chain_stat",
+      "min_chain_stat"
+    )
+  )
+  expect_named(
+    summary(epichains_tree2),
+    c(
+      "chains_ran",
+      "max_time",
+      "unique_ancestors",
+      "max_generation"
+    )
+  )
+  expect_named(
+    summary(epichains_tree2),
+    c(
+      "chains_ran",
+      "max_time",
+      "unique_ancestors",
+      "max_generation"
+    )
+  )
+  expect_true(
+    is.infinite(
+      summary(epichains_summary_all_infs)$min_chain_stat
+    )
+  )
+  expect_true(
+    is.infinite(
+      summary(epichains_summary_all_infs)$max_chain_stat
+    )
+  )
+})
+
+test_that("validate_epichains works", {
+  expect_invisible(
+    validate_epichains(epichains_summary)
+  )
+  expect_invisible(
+    validate_epichains(epichains_tree)
+  )
+  expect_invisible(
+    validate_epichains(epichains_tree2)
+  )
+})
+
+test_that("is_chains_tree works", {
+  expect_true(
+    is_chains_tree(epichains_tree)
+  )
+  expect_true(
+    is_chains_tree(epichains_tree2)
+  )
+  expect_false(
+    is_chains_tree(epichains_summary)
+  )
+})
+
+test_that("is_chains_summary works", {
+  expect_true(
+    is_chains_tree(epichains_tree)
+  )
+  expect_true(
+    is_chains_tree(epichains_tree2)
+  )
+  expect_false(
+    is_chains_tree(epichains_summary)
+  )
+})
```

Comment:
Here, you are testing `is_chains_tree()` where you probably intend `is_chains_summary()`

Reply:
The test is doing what's expected. I'm expecting `is_chains_tree()` to return FALSE on an object returned by `simulate_summary()`.

Reply:
But you have tested `is_chains_tree()` in the test immediately before this one - I would expect to see `expect_true(is_chains_summary(epichains_summary))` here, and an `expect_false()` on `is_chains_summary(epichains_tree)`.

Lines 111 - 121
```
test_that("is_chains_tree works", {
  expect_true(
    is_chains_tree(epichains_tree)
  )
  expect_true(
    is_chains_tree(epichains_tree2)
  )
  expect_false(
    is_chains_tree(epichains_summary)
  )
})
```

Reply:
Oh sorry. I see what you're saying now. I've fixed that. All functions in that test are now using `is_chains_summary()`. L343-L399.